Controller/AdminSecurityController.php

```diff
@@ -12,29 +12,29 @@
 namespace Sonata\UserBundle\Controller;
 
 use FOS\UserBundle\Model\UserInterface;
-use Symfony\Component\DependencyInjection\ContainerAware;
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
 use Symfony\Component\HttpFoundation\RedirectResponse;
 use Symfony\Component\HttpFoundation\Response;
 use Symfony\Component\Routing\Exception\RouteNotFoundException;
 use Symfony\Component\Security\Core\SecurityContext;
 
-class AdminSecurityController extends ContainerAware
+class AdminSecurityController extends Controller
 {
     /**
      * @return Response|RedirectResponse
      */
     public function loginAction()
     {
-        $user = $this->container->get('security.context')->getToken()->getUser();
+        $user = $this->get('security.context')->getToken()->getUser();
 
         if ($user instanceof UserInterface) {
-            $this->container->get('session')->getFlashBag()->set('sonata_user_error', 'sonata_user_already_authenticated');
-            $url = $this->container->get('router')->generate('sonata_admin_dashboard');
+            $this->get('session')->getFlashBag()->set('sonata_user_error', 'sonata_user_already_authenticated');
+            $url = $this->generateUrl('sonata_admin_dashboard');
 
-            return new RedirectResponse($url);
+            return $this->redirect($url);
         }
 
-        $request = $this->container->get('request');
+        $request = $this->get('request');
```

Comment:
Request should be passed on the method with null default value and a `resolveRequest` should be written for BC.